### DISCUSSION TITLE
can you automate pin pairing sunshine/moonlight? sunshine flags -0 | Read PIN from stdin

### DISCUSSION THREAD

**Discussion by comsyspro** [LizardByte/community#292] [Sunshine] [answered]
for an automated setup of sunshine and moonlight i want to ask if you can use this flag `-0 | Read PIN from stdin` somehow to input a pin. on the moonlight side you can also set the pairing with this command `moonlight.exe --pin 1234 pair ip:port`. the problem now is how can you set the flag on sunshine side for stdin the pin? i tried this `sunshine.exe credentials_file=sunshine_state1.json -0 1234` and also  this `sunshine.exe -0 1234 credentials_file=sunshine_state1.json` but the pin is not recognized.

here is the sunshine help:
```
Usage: sunshine.exe [options] [/path/to/configuration_file] [--cmd]
    Any configurable option can be overwritten with: "name=value"

    Note: The configuration will be created if it doesn't exist.

    --help                    | print help
    --creds username password | set user credentials for the Web manager
    --version                 | print the version of sunshine

    flags
        -0 | Read PIN from stdin
        -1 | Do not load previously saved state and do retain any state after shutdown
           | Effectively starting as if for the first time without overwriting any pairings with your devices
        -2 | Force replacement of headers in video stream
        -p | Enable/Disable UPnP
```
how to use the flags here?
https://github.com/LizardByte/Sunshine/blob/31e8b798dabf47d5847a1b485f57cf850a15fcae/src/config.h#L118

or is it also possible to set a constant/fixed pin?
https://github.com/LizardByte/Sunshine/blob/31e8b798dabf47d5847a1b485f57cf850a15fcae/src/config.h#L122

**Comment by ReenigneArcher** [comment, marked as answer]
Sorry for the late reply... I don't think anyone is using the STDIN pin since we implemented the Web UI, which has been about 2.5 years now.

Just a guess, but it might just be `sunshine.exe -0` ... and wait for it to tell you to enter a pin?

> or is it also possible to set a constant/fixed pin?

No.

**Reply by oyeakhill** [reply]
Hi. Is there another way to bypass pin restrictions through web UI?
Do we have to change something in sunshine config? If you could guide

**Reply by ReenigneArcher** [reply]
No.
